system-test/storage.ts

```diff
@@ -1217,6 +1217,66 @@ describe('storage', () => {
     });
   });
 
+  describe('cors configuration', () => {
+    const corsEntry = [
+      {
+        maxAgeSeconds: 1600
+      },
+      {
+        maxAgeSeconds: 3600,
+        method: [
+          "GET",
+          "POST"
+        ],
+        origin: ["*"],
+        responseHeader: [
+          "Content-Type",
+          "Access-Control-Allow-Origin"
+        ]
+    }];
+
+    describe('bucket', () => {
+      it('should create a bucket with a CORS configuration when passed in', async () => {
+        const bucket = storage.bucket(generateName());
+        await storage.createBucket(bucket.name, {
+          cors: corsEntry,
+        });
+
+        await bucket.getMetadata();
+        assert.strictEqual(
+          bucket.metadata.cors,
+          corsEntry
+        );
+      })
+
+      it('should set a CORS configuration', async () => {
+        const bucket = storage.bucket(generateName());
+        await bucket.create();
+        await bucket.setCorsConfiguration(corsEntry);
+        await bucket.getMetadata();
+        assert.strictEqual(
+          bucket.metadata.cors,
+          corsEntry
+        );        
+      });
+
+      it('should remove a CORS configuration', async () => {
+        const bucket = storage.bucket(generateName());
+        await bucket.create();
+        await bucket.setCorsConfiguration(corsEntry);
+        await bucket.getMetadata();
+        assert.strictEqual(
+          bucket.metadata.cors,
+          corsEntry
+        );
+        
+        // And now test the removing
+        bucket.setCorsConfiguration([]);
+        assert.strictEqual(bucket.metadata.cors.length, 0);
```

Comment:
after removing the CORS configs `bucket.metadata.cors` will be `undefined`
```suggestion
        assert.ok(!bucket.metadata.cors);
```

Reply:
I see that now; thanks!